crates/dbsp/src/trace/ord/file/val_batch.rs

```diff
@@ -1031,26 +1084,37 @@ where
     R: WeightTrait + ?Sized,
 {
     fn push_time(&mut self, key: &K, val: &V, time: &T, weight: &R) {
-        if let Some(cur_key) = self.cur_key.get() {
-            if key != cur_key {
-                self.writer.write0((cur_key, ().erase())).unwrap();
-                self.cur_key.from_ref(key);
+        if let Some(cur) = &mut self.cur {
```

Comment:
Let's add a `debug_assert` here that weight isn't zero.

Reply:
You mean like this?

```patch
diff --git a/crates/dbsp/src/trace/ord/file/val_batch.rs b/crates/dbsp/src/trace/ord/file/val_batch.rs
index b6584a0a3..04cb787e8 100644
--- a/crates/dbsp/src/trace/ord/file/val_batch.rs
+++ b/crates/dbsp/src/trace/ord/file/val_batch.rs
@@ -1084,6 +1084,7 @@ where
     R: WeightTrait + ?Sized,
 {
     fn push_time(&mut self, key: &K, val: &V, time: &T, weight: &R) {
+        debug_assert!(!weight.is_zero());
         if let Some(cur) = &mut self.cur {
             if &*cur.key != key {
                 // Key differs from previous, so write the old (V, timediffs) to
```